src/App.tsx

```diff
@@ -1,9 +1,45 @@
-import React from 'react';
+import {
+  FC,
+  useState,
+  BaseSyntheticEvent,
+} from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App: React.FC = () => {
+interface Movie {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+}
+
+const getLowerCase = (expression: string) => (
+  expression.toLocaleLowerCase()
+);
+
+const handleQueryChange = (query: string): Movie[] => {
+  const queryLowerCase = getLowerCase(query);
+
+  return moviesFromServer.filter((movie: Movie) => (
+    getLowerCase(movie.title)
+      .includes(queryLowerCase)
+    || getLowerCase(movie.description)
+      .includes(queryLowerCase)
```

Comment:
looks like you duplicate yourself with these pieces of code
```
getLowerCase(...).includes(...)
```

Reply:
also, if to talk about time complexity (if you don't know about it – google, suggest learning it a little bit more):
```
Array.filter { // O(N)
  String.toLocaleLowerCase // O(N)
  String.includes // O(N)
}

// it can be represented this way – O(N) * (O(N) + O(N)) = 2 * O(N)² ≈ O(N)²
```
and that's a time-consuming operation, try to optimize it a bit if you want, it is not obligatory